dataprovider/dataprovider.go

```diff
@@ -471,7 +471,11 @@ func (provider *DataProvider) handleSubscribeChannel(adapter DataAdapter) {
 
 		for path, value := range changes {
 			for idElement := provider.sensors[path].subscribeIds.Front(); idElement != nil; idElement = idElement.Next() {
-				id := idElement.Value.(uint64)
+				id, ok := idElement.Value.(uint64)
+				if !ok {
+					log.Error("Incorrect value format")
```

Comment:
More informative log is required:
```go
log.Error("Wrong subscribe ID type")
```
or something like that.

Reply:
done